### PR TITLE
Fix eslint issues

### PR DESCRIPTION
**Pull request by renatho**
## Description

This PR fixes some Eslint issues related to PR #25300.
It still contains some issues related to tests (tests without assertion and jasmine global - see screenshots) which I preferred don't touch for now because we could follow different approaches and I'm not sure what would fit better.

I separated the changes in different commits, where the first one is about indentation only. In the second one, I also added some arguments that were missing with their types. I identified the type through the code interpretation, so I'd appreciate a more detailed check here: https://github.com/WordPress/gutenberg/pull/25352/commits/598aea5bd59c9207ba5fb8aadf4d211ec721c955

## How has this been tested?

Running `npm run lint-js` in Gutenberg code.

## Screenshots 

The remaining issues (warnings that were present before the changes too):

Screen Shot 2020-09-15 at 19 33 52

## Types of changes

Eslint fixes related to JSDoc.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/env/lib/config/config.js [hunk 1]
```diff
@@ -19,25 +19,25 @@ const md5 = require( '../md5' );
...
- * @property {string}                           dockerComposeConfigPath Path to the docker-compose.yml file.
- * @property {boolean}                          detectedLocalConfig     If true, wp-env detected local config and used it.
- * @property {Object.<string, WPServiceConfig>} env                     Specific config for different environments.
- * @property {boolean}                          debug                   True if debug mode is enabled.
+ * @property {string}         name                    Name of the environment.
+ * @property {string}         configDirectoryPath     Path to the .wp-env.json file.
+ * @property {string}         workDirectoryPath       Path to the work directory located in ~/.wp-env.
+ * @property {string}         dockerComposeConfigPath Path to the docker-compose.yml file.
+ * @property {boolean}        detectedLocalConfig     If true, wp-env detected local config and used it.
+ * @property {Object.<string, WPServiceConfig>}       env                     Specific config for different environments.
```
Comment (noahtallen) [inline packages/env/lib/config/config.js:27]: this seems like a weird fix. Why it is aligning the variables with the type parameter? 🤔

**Review by noahtallen — APPROVED** [review]
This looks fine just looking at `/components/env`

File: packages/env/lib/config/config.js (hunk 1 above)
Comment (renatho) [reply packages/env/lib/config/config.js]: Good catch @noahtallen! It's an issue with the plugin that I haven't seen before. For now, I just updated removing the space, but I'll submit a PR to the `eslint-jsdoc-plugin` with a fix in the next few days.

File: bin/packages/build.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ const PACKAGES_DIR = path.resolve( __dirname, '../../packages' );
 /**
  * Get the package name for a specified file
  *
- * @param  {string} file File name
+ * @param {string} file File name
  * @return {string}      Package name
```
Comment (ntwb) [inline bin/packages/build.js:23]: If `@return` descriptions are not destined to be aligned with `@param` descriptions then I think this also requires some changes:

This aligns the descriptions of both lines:
```suggestion
 * @param {string} file File name
 * @return {string}     Package name
```

Or, this removes the additional whitespece preceding the @return description

```diff
 * @param {string} file File name
 * @return {string} Package name
```

Possibly, another convention observed further below is a blank line between the last @param and the @raturn:

```diff
 * @param {string} file File name
 *
 * @return {string} Package name
```

File: bin/plugin/config.js [hunk 3]
```diff
@@ -3,20 +3,20 @@ const gitRepoOwner = 'WordPress';
...
- * @property {string} name Name.
- * @property {string} team Github Team Name.
- * @property {string} versionMilestoneFormat printf template for milestone
+ * @property {string} slug                    Slug.
+ * @property {string} name                    Name.
+ * @property {string} team                    Github Team Name.
+ * @property {string} versionMilestoneFormat  printf template for milestone
  *                                           version name. Expected to be called
  *                                           with a merged object of the config
  *                                           and semver-parsed version parts.
```
Comment (ntwb) [inline bin/plugin/config.js:12]:
```suggestion
 * @property {string} versionMilestoneFormat  printf template for milestone
 *                                            version name. Expected to be called
 *                                            with a merged object of the config
 *                                            and semver-parsed version parts.
```

Identation of the subsequent lines should align with the start of the first comment

File: bin/plugin/lib/git.js [hunk 4]
```diff
@@ -47,7 +47,7 @@ async function commit( gitWorkingDirectoryPath, message, filesToAdd = [] ) {
  * Creates a local branch.
  *
  * @param {string} gitWorkingDirectoryPath Local repository path.
- * @param {string} branchName Branch Name
+ * @param {string} branchName              Branch Name
```
Comment (ntwb) [inline bin/plugin/lib/git.js:50]: Not a blocker, but we should look at standardizing either including or excluding the `.` period at the end of the @param description, PHP Coding Standards require this, so I'd lean toward that also for JS (A goal is to align the CS between languages as best we can when suited)

File: packages/annotations/src/format/annotation.js [hunk 5]
```diff
@@ -12,8 +12,8 @@ const STORE_KEY = 'core/annotations';
 /**
  * Applies given annotations to the given record.
  *
- * @param {Object} record The record to apply annotations to.
- * @param {Array} annotations The annotation to apply.
+ * @param {Object} record      The record to apply annotations to.
+ * @param {Array}  annotations The annotation to apply.
  * @return {Object} A record with the annotations applied.
```
Comment (ntwb) [inline packages/annotations/src/format/annotation.js:17]: FYI Only: An example where there's no blank line between the last @param and the @return and the @return description is not aligned with the @param descriptions

File: packages/autop/src/index.js [hunk 6]
```diff
@@ -123,8 +123,8 @@ function replaceInHtmlTags( haystack, replacePairs ) {
  * replace double line-breaks with HTML paragraph tags. The remaining line-
  * breaks after conversion become `<br />` tags, unless br is set to 'false'.
  *
- * @param  {string}    text The text which has to be formatted.
- * @param  {boolean}   br   Optional. If set, will convert all remaining line-
+ * @param {string}  text The text which has to be formatted.
+ * @param {boolean} br   Optional. If set, will convert all remaining line-
  *                          breaks after paragraphing. Default true.
```
Comment (ntwb) [inline packages/autop/src/index.js:128]:
```suggestion
 * @param {boolean} br   Optional. If set, will convert all remaining line-
 *                       breaks after paragraphing. Default true.
```

The original second line aligned with the start of the original description, seems like this should be updated here also

File: packages/block-editor/src/autocompleters/block.js [hunk 7]
```diff
@@ -38,11 +38,6 @@ const createBlocksFromInnerBlocksTemplate = ( innerBlocksTemplate ) => {
 /**
  * Creates a blocks repeater for replacing the current block with a selected block type.
  *
- * @param {Object} props                                   Component props.
- * @param {string} [props.getBlockInsertionParentClientId] Client ID of the parent.
- * @param {string} [props.getInserterItems]                Inserter items for parent.
- * @param {string} [props.getSelectedBlockName]            Name of selected block or null.
- *
```
Comment (ntwb) [inline packages/block-editor/src/autocompleters/block.js:45]: Can this be confirmed these should be removed please?

File: packages/block-editor/src/components/block-navigation/use-block-navigation-drop-zone.js [hunk 8]
```diff
@@ -38,10 +38,10 @@ import useOnBlockDrop from '../use-on-block-drop';
...
  * @typedef {Object} WPBlockNavigationDropZoneTarget
- * @property {string}                   blockIndex   The insertion index.
- * @property {string}                   rootClientId The root client id for the block.
- * @property {string|undefined}         clientId     The client id for the block.
- * @property {'top'|'bottom'|'inside'}  dropPosition The position relative to the block that the user is dropping to.
+ * @property {string}                  blockIndex   The insertion index.
+ * @property {string}                  rootClientId The root client id for the block.
+ * @property {string|undefined}        clientId     The client id for the block.
+ * @property {'top'|'bottom'|'inside'} dropPosition The position relative to the block that the user is dropping to.
  *                                                   'inside' refers to nesting as an inner block.
```
Comment (ntwb) [inline packages/block-editor/src/components/block-navigation/use-block-navigation-drop-zone.js:45]:
```suggestion
 * @property {'top'|'bottom'|'inside'} dropPosition The position relative to the block that the user is dropping to.
 *                                                  'inside' refers to nesting as an inner block.
```

Second line indent update

File: packages/block-editor/src/components/inner-blocks/use-inner-block-template-sync.js [hunk 9]
```diff
@@ -17,9 +17,9 @@ import { synchronizeBlocksWithTemplate } from '@wordpress/blocks';
  * is meant to be locked (e.g. templateLock = "all"), then we replace the inner
  * blocks with the correct value after synchronizing it with the template.
  *
- * @param {string} clientId     The block client ID.
- * @param {Object} template     The template to match.
- * @param {string} templateLock The template lock state for the inner blocks. For
+ * @param {string}  clientId                       The block client ID.
+ * @param {Object}  template                       The template to match.
+ * @param {string}  templateLock                   The template lock state for the inner blocks. For
```
Comment (ntwb) [inline packages/block-editor/src/components/inner-blocks/use-inner-block-template-sync.js:22]: This needs updating and the lines below to line 31, looks like autofix missed the `@param` on line 28

File: packages/block-editor/src/components/inserter/hooks/use-block-types-state.js [hunk 10]
```diff
@@ -26,8 +26,8 @@ const createBlocksFromInnerBlocksTemplate = ( innerBlocksTemplate ) => {
 /**
  * Retrieves the block types inserter state.
  *
- * @param {string=}  rootClientId        Insertion's root client ID.
- * @param {Function} onInsert            function called when inserter a list of blocks.
+ * @param {string=}  rootClientId Insertion's root client ID.
+ * @param {Function} onInsert     function called when inserter a list of blocks.
```
Comment (ntwb) [inline packages/block-editor/src/components/inserter/hooks/use-block-types-state.js:30]:
```suggestion
 * @param {Function} onInsert     Function called when inserter a list of blocks.
```

Should descriptions always begin with a Capital letter? It appears to be the convention and is also a PHP coding standard

File: packages/block-editor/src/components/provider/use-block-sync.js [hunk 11]
```diff
@@ -40,25 +40,25 @@ import { useRegistry } from '@wordpress/data';
  *   controllers.
```
Comment (ntwb) [inline packages/block-editor/src/components/provider/use-block-sync.js:40]: More examples in this file of second line descriptions not aligned with the first line

File: packages/block-editor/src/hooks/align.js [hunk 12]
```diff
@@ -80,7 +80,7 @@ export function getValidAlignments(
 /**
  * Filters registered block settings, extending attributes to include `align`.
  *
- * @param  {Object} settings Original block settings
+ * @param {Object} settings Original block settings
  * @return {Object}          Filtered block settings
```
Comment (ntwb) [inline packages/block-editor/src/hooks/align.js:84]:
```suggestion
 * @return {Object} Filtered block settings
```

Remove additional whitespace for @return descriptions

File: packages/block-editor/src/store/reducer.js [hunk 13]
```diff
@@ -138,9 +138,9 @@ function getFlattenedBlockAttributes( blocks ) {
  * an inner block controller. To do so, it must be excluded from the list of
  * client IDs which are considered to be part of the top-level entity.
  *
- * @param {Object}  blocksOrder  Object that maps block client IDs to a list of
+ * @param {Object}  blocksOrder           Object that maps block client IDs to a list of
  *                               nested block client IDs.
```
Comment (ntwb) [inline packages/block-editor/src/store/reducer.js:142]:
```suggestion
 * @param {Object}  blocksOrder           Object that maps block client IDs to a list of
 *                                        nested block client IDs.
```

**Review by ntwb — CHANGES_REQUESTED** [review]
Thanks, left a few comments, as we are _touching_ these lines now, helping our future selves for Git blame, maybe adding the trailing `.` period to the `@param` descriptions now that are missing them would be good (It appears there are not many instances where these are missing)

I got through around ~25% of the files, can we make those changes noted in the review and then I can come back for a full review pass of all the files

Edit: It may in fact we worthwhile trying to use Prettier prettier-plugin-jsdoc here to update the docs s mentioned here in https://github.com/WordPress/gutenberg/issues/24984#issuecomment-690572969 as this may be signifantly quicker for both this PR and future code written will be formatted automatically rather than a devloper pushing up code that then fails linting

**Comment by renatho** [comment]
Thank you so much for the review @ntwb!!

As I mentioned here: https://github.com/WordPress/gutenberg/pull/25300#issuecomment-693615273, I think I'll keep this PR in stand by for now until going with that fixes.

But it worthed a lot to get some more cases and maybe include some of them in the rule logic.

I'm closing this for now and I'll open a new one when it's ready. It'll be easier because it'll have a lot of conflicts.

**Pull request closed without merging**